js/main.js

```diff
@@ -724,3 +733,68 @@ function renderMyCities() {
 function clearMessage() {
   $modalMessage.textContent = '';
 }
+
+function sortMyCities(event) {
+  if (event.target.value === 'recent') {
+    data.myEntries = data.myEntries.sort(function (a, b) {
+      if (a.visitDate > b.visitDate) {
+        return -1;
+      } else if (a.visitDate < b.visitDate) {
+        return 1;
+      } else {
+        return 0;
+      }
+    });
+  } else if (event.target.value === 'oldest') {
+    data.myEntries = data.myEntries.sort(function (a, b) {
+      if (a.visitDate < b.visitDate) {
+        return -1;
+      } else if (a.visitDate > b.visitDate) {
+        return 1;
+      } else {
+        return 0;
+      }
+    });
+  } else if (event.target.value === 'city') {
+    data.myEntries = data.myEntries.sort(function (a, b) {
+      if (a.cityName.toLowerCase() > b.cityName.toLowerCase()) {
+        return 1;
+      } else if (a.cityName.toLowerCase() < b.cityName.toLowerCase()) {
+        return -1;
+      } else {
+        return 0;
+      }
+    });
+  } else if (event.target.value === 'rev-city') {
+    data.myEntries = data.myEntries.sort(function (a, b) {
+      if (a.cityName.toLowerCase() < b.cityName.toLowerCase()) {
+        return 1;
+      } else if (a.cityName.toLowerCase() > b.cityName.toLowerCase()) {
+        return -1;
+      } else {
+        return 0;
+      }
+    });
+  } else if (event.target.value === 'country') {
+    data.myEntries = data.myEntries.sort(function (a, b) {
+      if (a.cityCountry.toLowerCase() > b.cityCountry.toLowerCase()) {
+        return 1;
+      } else if (a.cityCountry.toLowerCase() < b.cityCountry.toLowerCase()) {
+        return -1;
+      } else {
+        return 0;
+      }
+    });
+  } else if (event.target.value === 'rev-country') {
+    data.myEntries = data.myEntries.sort(function (a, b) {
+      if (a.cityCountry.toLowerCase() < b.cityCountry.toLowerCase()) {
+        return 1;
+      } else if (a.cityCountry.toLowerCase() > b.cityCountry.toLowerCase()) {
+        return -1;
+      } else {
+        return 0;
+      }
+    });
```

Comment:
I would make each of these anonymous functions into a named function that returns the array sorted in the order it should be.